src/anomalib/engine/engine.py

```diff
@@ -213,6 +267,46 @@ def _setup_anomalib_callbacks(self) -> None:
             self.trainer.callbacks + _callbacks,
         )
 
+    def _should_run_validation(
+        self,
+        model: AnomalyModule,
+        dataloaders: EVAL_DATALOADERS | AnomalibDataModule | None,
+        datamodule: AnomalibDataModule | None,
+        ckpt_path: str | None,
+    ) -> bool:
+        """Check if we need to run validation to collect normalization statistics and thresholds.
+
+        If a checkpoint path is provided, we don't need to run validation because we can load the model from the
+        checkpoint and use the normalization metrics and thresholds from the checkpoint.
+
+        We need to run validation if the model is configured with normalization enabled, but no normalization metrics
+        have been collected yet. Similarly, we need to run validation if the model is configured with adaptive
+        thresholding enabled, but no thresholds have been computed yet.
+
+        We can only run validation if we have validation data available, so we check if the dataloaders or datamodule
+        are available. If neither is available, we can't run validation.
+
+        Args:
+            model (AnomalyModule): Model passed to the entrypoint.
+            dataloaders (EVAL_DATALOADERS | AnomalibDataModule | None): Dataloaders passed to the entrypoint.
```

Comment:
I guess we'll address this `AnomalibDataModule` annotation later, as we discussed. It looks a bit confusing.